pkg/controllers/cluster/cluster_controller.go

```diff
@@ -311,24 +304,26 @@ func (c *Controller) removeExecutionSpace(cluster *clusterv1alpha1.Cluster) erro
 		return err
 	}
 
-	return nil
-}
-
-// ExecutionSpaceExistForCluster determine whether the execution space exists in the cluster
-func (c *Controller) ExecutionSpaceExistForCluster(clusterName string) (bool, error) {
-	executionSpaceName := names.GenerateExecutionSpaceName(clusterName)
-
-	executionSpaceObj := &corev1.Namespace{}
-	err := c.Client.Get(context.TODO(), types.NamespacedName{Name: executionSpaceName}, executionSpaceObj)
-	if apierrors.IsNotFound(err) {
-		klog.V(2).Infof("Execution space(%s) no longer exists", executionSpaceName)
+	// make sure the execution space has been deleted
+	err := wait.Poll(1*time.Second, 10*time.Second, func() (done bool, err error) {
```

Comment:
Sorry, I don't understand why your solution for this is better than the current one?

Reply:
We can refer to this analysis process: https://github.com/karmada-io/karmada/issues/3113#issuecomment-1422124843

I'm not sure how well the optimized logic improves in real-world testing, hi @yanfeng1992, can you help make a test?